### DISCUSSION TITLE
VueRouter render issue with NuxtLayout as root (SPA)

### DISCUSSION THREAD

**Discussion by angelhdzmultimedia** [nuxt/nuxt#25477] [Questions] [answered]
Hello!

This issue was driving me insane! I downgraded and upgraded to many versions of vue and vue-router, believing they were the cause because I have always used `NuxtLayout` as root element inside the template of each SFC.

When navigating to a route either with `navigateTo` or with `<NuxtLink to="...">`, the address bar url changes, but not the content.
I even used a `:key="$route.path"` in the `~/app.vue` `NuxtPage`.
So today had a feeling and wrapped each page NuxtLayout with a div and BAM! It worked right away.

### Question...
What changed?

**Comment by TheAlexLichter** [comment, marked as answer]
See https://github.com/nuxt/nuxt/issues/25214
